src/filesystem.cpp

```diff
@@ -69,18 +201,24 @@ void FileSystemAccess::escapefsincompatible(string* name) const
     }
 
     char buf[4];
-    unsigned char c;
-
-    // replace all occurrences of a badchar with %xx
-    for (size_t i = name->size(); i--; )
-    {
-        c = (unsigned char)(*name)[i];
-
-        if (!islocalfscompatible(c))
+    int fileSystemType = getlocalfstype(&validPath);
+    size_t utf8seqsize = 0, i = 0;
+    unsigned char c = '0';
+    while (i < name->size())
+    {
+        c = static_cast<unsigned char>((*name)[i]);
+        utf8seqsize = Utils::utf8SequenceSize(c);
+        assert (utf8seqsize);
+        if (utf8seqsize == 1 && !islocalfscompatible(c, fileSystemType))
         {
+            std::string inc = name->substr(i, 1);
```

Comment:
I think `inc` should be declared as a `char` (and with a name like `incompatibleChar` or alike)

Reply:
You can simply replace the `inc` in the log by `name->substr(i, 1);`

Reply:
I've forgotten the reason behind std::string inc. name is modified so I need to store the current value and I can't  replace in the log by name->substr(i, 1);